### DISCUSSION TITLE
How do I deal with flicker along the edges of a TextureAtlasSprite when it/the camera is moving?

### DISCUSSION THREAD

**Discussion by emctague** [bevyengine/bevy#8618] [Q&A]
Preview video:

https://github.com/bevyengine/bevy/assets/22970065/ab262e01-42b8-4c21-a9ee-52b85e5c6b1f

I assume it's sampling a bit of the white from the tile above it, but I'm not sure how to fix it other than ripping the character sprites right out of the tilemap (I didn't make the tilemap). Is there an in-engine solution?

**Comment by DrAklinger** [comment]
How do you create your TextureAtlas which is feeding the TextureAtlasSprite? Is it via loading the animation images into a TextureAtlas or by creating a TextureAtlas via the 'from_grid' method? 

If you do the second one, maybe you missed the setting of some padding pixels in-between the different frames? Or the offset is off by a pixel?

**Reply by emctague** [reply]
I'm currently using [bevy_ecs_ldtk](https://github.com/Trouv/bevy_ecs_ldtk)'s automatic loading method, which is essentially loading it from a grid internally. I used from_grid before that, though, with the same outcome.

The texture I am loading from has no spacing between sprites, though, and I was under the impression that the "padding" parameter allowed for the TextureAtlas to handle the case of an n-pixel gap between each sprite in the image file?

**Reply by DrAklinger** [reply]
Yes, the padding would allow for in-between space. 

Maybe the flickering is a result of z fighting? Are there other materials on the same z layer? I also got some flickering when I had 'nothing' behind a sprite. Maybe putting something further in the back (sky box thingi) could also help.

**Reply by mabysh** [reply]
@emctague Have you found any solution to this? I'm experiencing the same issue in bevy 0.12.1.
